File: BTVN3/bai11.py
class Item:
    def __init__(self, weight: int, value: int):
        self.weight = weight
        self.value = value

def solve() -> None:
    n, x = map(int, input().split())
    items = [Item(*map(int, input().split())) for _ in range(n)]
    dp = [0 for _ in range(x + 1)]
    for i in range(n):
        for j in range(x, 0, -1):
            if items[i].weight <= j:
                dp[j] = max(dp[j], dp[j-items[i].weight] + items[i].value)

    print(dp[x])

if __name__ == "__main__":
    solve()

File: BTVN3/bai11.cpp
#include<iostream>
#include<vector>
#include<algorithm>
using namespace std;
struct Item {
    int weight, value;
};
int main() {
    int n, x;
    cin >> n >> x;
    vector<Item> I(n+1);
    for(int i = 0; i < n; i++) {
        cin >> I[i].weight >> I[i].value;
    }
    vector<long long> dp(x + 1, 0);
    for(int i = 0; i < n; i++) {
        for(int j = x; j >= 1; j--) {
            if(I[i].weight <= j)
                dp[j] = max(dp[j], dp[j-I[i].weight] + I[i].value);
            else continue;
        }
    }
    cout << dp[x];
    return 0;
}